synapse/logging/_terse_json.py

```diff
@@ -188,36 +231,36 @@ def start(self) -> None:
         self._service = ClientService(endpoint, factory, clock=self.hs.get_reactor())
         self._service.startService()
 
+    def stop(self):
+        self._service.stopService()
+
     def _write_loop(self) -> None:
         """
         Implement the write loop.
         """
         if self._writer:
             return
 
+        if self._producer and self._producer._transport.connected:
+            self._producer.resumeProducing()
+            return
+
         self._writer = self._service.whenConnected()
 
-        @self._writer.addBoth
+        @self._writer.addErrback
+        def fail(r):
+            r.printTraceback(file=sys.__stderr__)
+            self._writer = None
+            self.hs.get_reactor().callLater(1, self._write_loop)
```

Comment:
Should we just have a single looping call that calls this once a second? If `_write_loop` is async then it should do the right thing. and avoids the hit of creating temporary functions all the time